src/kernel/include/shared/cos_types.h

```diff
@@ -40,7 +50,16 @@ typedef u32_t         sched_tok_t;
  */
 #define TCAP_TIME_QUANTUM_ORD 12
 #define TCAP_TIME_MAX_ORD (TCAP_TIME_QUANTUM_ORD + (sizeof(tcap_time_t) * 8))
-#define TCAP_TIME_MAX_BITS(c) ((c >> TCAP_TIME_MAX_ORD) << TCAP_TIME_MAX_ORD)
+/*
+ * FIXME
+ * TCAP_TIME_MAX_ORD comes out as 44, thus shifting by this much on a 32 bit
+ * system leads to warnings that shift is greater than the width of the type.
+ * As we shift right by this amount, it zeros out the number. To avoid
+ * warnings I have temporarily replaced this initial shift by 32 bits.
+ * Fix when I know the original purpose of this macro.
+ */
+//#define TCAP_TIME_MAX_BITS(c) ((c >> TCAP_TIME_MAX_ORD) << TCAP_TIME_MAX_ORD)
+#define TCAP_TIME_MAX_BITS(c) ((c >> 32) << 32)
```

Comment:
@phanikishoreg you need to figure out the solution to this.  I don't remember the context.